contracts/Committee.sol

```diff
@@ -310,135 +314,113 @@ contract Committee is ICommittee, ContractRegistryAccessor, WithClaimableFunctio
 	}
 
 	function _rankAndUpdateMember(Member memory member) private returns (bool committeeChanged, bool standbysChanged) {
-		(committeeChanged, standbysChanged) = _rankMember(member);
-		membersData[member.addr] = member.data;
-	}
-
-	function _rankMember(Member memory member) private returns (bool committeeChanged, bool standbysChanged) {
-		Settings memory _settings = settings;
-
-		bool isParticipant = member.data.inCommittee || member.data.isStandby;
-		if (!isParticipant && !qualifiesAsStandby(member)) {
+		if (!isCommitteeMemberOrStandby(member.data) && !qualifiesAsStandby(member.data)) {
+			membersData[member.addr] = member.data;
 			return (false, false);
 		}
-
-		return updateCommittee(member, _settings);
-	}
-
-	function isReadyToSyncStale(uint48 timestamp, bool currentlyInCommittee, Settings memory _settings) private view returns (bool) {
-		return timestamp == 0 || !currentlyInCommittee && timestamp <= uint48(now) - _settings.readyToSyncTimeout;
+		(committeeChanged, standbysChanged) = updateCommittee(member, settings);
+		membersData[member.addr] = member.data;
 	}
 
-	function qualifiesAsStandby(Member memory member) private pure returns (bool) {
-		return member.data.isMember && member.data.readyToSyncTimestamp != 0 && member.data.weight != 0;
+	function isTimedOut(Participant memory p, Settings memory _settings) private view returns (bool) {
+		return p.data.readyToSyncTimestamp == 0 || (p.oldRole != ROLE_COMMITTEE && p.data.readyToSyncTimestamp <= uint48(now) - _settings.readyToSyncTimeout);
```

Comment:
why is `p.data.readyToSyncTimestamp == 0 ||` needed?

Reply:
`p.data.readyToSyncTimestamp == 0` means you are explicitly not ready (never sent `readyToSync` or by `memberNotReadyToSync`).
Otherwise, we check the timeout only if you are not currently a committee member.